### DISCUSSION TITLE
I need to group resources somehow

### DISCUSSION THREAD

**Discussion by vitaly107** [avo-hq/avo#510] [Help]
Say I want to display only links to resources in a certain group.

How can I tag/label a resource and then pick only resources with that label?

Or just group them somehow and then pick up a bunch by a group name.

Many thanks

**Comment by adrianthedev** [comment]
Hi Vitaly,

Not sure I see what you mean.

Like if you have a `Post` resource you'd like to have two pages displaying posts? On one page some posts and on another, other posts.
Something like that?

**Reply by vitaly107** [reply]
I want to display links to certain groups of resources/models on different custom dashboards, for example on customer services dashboard I want to display Users, addresses and something else and on Marketing dashboard I want to display Page, Foter Options and so on.

so I need a way to group resources and then iterate only through a specific group **to create navigation links** to resources in that group

**Comment by vitaly107** [comment]
in other words instead of showing all the resources using 
```
 <%# Avo::App.resources_navigation(_current_user).each do |resource| %>
            <%#= render Avo::NavigationLinkComponent.new label: resource.name.pluralize.humanize, path: resources_path(resource.model_class) %>
          <%# end %>
```
I want to show/iterate ove one group only in one page and a different group on a different page.

So I need to do something that go into a particular resource and label it as 'Marketing Stuff` and the a different resource as 'Customer stuff`

If there no way to label or tag as yet then please give me a clue how to groupd them and then refer to the group / iterate of a group of resources not all of them. Cheers!

**Comment by vitaly107** [comment]
In other words how to use this

```
 <% Avo::App.resources_navigation(_current_user).each do |resource| %>
            <%= render Avo::NavigationLinkComponent.new label: resource.name.pluralize.humanize, path: resources_path(resource.model_class) %>
          <% end %>

only with certain group of resources?
```

**Comment by adrianthedev** [comment]
So, if you haven't already, you should eject the sidebar view to take control of the items shown in the sidebar. Then you create the rules you want.

If you do this `items = Avo::App.resources_navigation(_current_user)` you then have an array of sidebar items that you can rename/rearrange/hide & show.
You can transform that array into a hash and put it back in the sidebar.

Then you can add them back to your sidebar like so:
```ruby
items.each do |label, resource| %>
  <%= render Avo::NavigationLinkComponent.new label: label, path: resources_path(resource.model_class) %>
<% end %>
```

So in essence you should do something like this:

```ruby
items = Avo::App.resources_navigation(_current_user)

# manipulate items while transforming it into a hash with the desired label
# ex:
items = items.map do |resource|
  if resource.name === 'User" && SOME_OTHER_CONDITION
	return ['Marketing', resource]
  end
  [resource.name.pluralize.humanize, resource]
end.to_h

items.each do |label, resource| %>
  <%= render Avo::NavigationLinkComponent.new label: label, path: resources_path(resource.model_class) %>
<% end %>
```

**Reply by vitaly107** [reply]
Thanks

**Reply by adrianthedev** [reply]
Sure! Glad to be of help!
